hooks/pre-push

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/bash
+
+set -e
+
+flutter format -o none --set-exit-if-changed .
+flutter analyze .
+flutter test -r expanded .
```

Comment:
Are we sure we want to run the tests as well?
This takes quite some time and they will be run in the CI workflow regardless.

Reply:
I think we should as running the tests beforehand can prevent the extra commits made for passing all the checks.

Reply:
I was hoping we could start soon on adding testing on all business and application logic, even if this does not take a long time to run now, it might soon be so slow we have to remove it.

Reply:
okay 👍🏼